packages/playground/src/components/profile_manager/ConnectWallet.vue

```diff
@@ -318,7 +318,29 @@ const validateMnemonicInput = async (input: string) => {
 
   try {
     const grid = await getGrid({ mnemonic: mnemonic.value, keypairType: keypairType.value });
-    getEmail(grid!);
+
+    // Check if twin exists first
+    try {
+      await grid!.twins.get_my_twin_id();
+      // Twin exists, so get email with loading state
+      loadEmail.value = true;
+      try {
+        getEmail(grid!);
+      } finally {
+        loadEmail.value = false;
+      }
```

Comment:
why did you remove the load email handler from getEmail? 
also the old logic was handle getEmail errors, so no need to handle it here in nested try

Reply:
<img width="483" height="296" alt="image" src="https://github.com/user-attachments/assets/6bf05678-6a79-475f-86fb-3b7c3c69e558" />

i don't think it was firing any errors. 
also in your new code `await grid!.twins.get_my_twin_id();` if this failed so it will not trigger load email as it will go to the catch block